ui/plugins/auto_classification/controller.js

```diff
@@ -69,891 +40,1104 @@ treeherder.factory('thStringOverlap', function() {
-                                               getClassificationMatches);
-
-            // add a manual option (for manually inputting a bug associated
-            // with the failure line)
-            ui.options.push(new ThClassificationOption("manual", "manual"));
+        /**
+         * Determine whether the line should be open for editing by default
+         */
+        function defaultEditable() {
+            var option = currentOption();
+            return !(option.score >= goodMatchScore || option.type === "ignore");
+        }
 
-            // add an ignore option (for automatically ignoring the failure
-            // line in the future)
-            var ignoreOption = new ThClassificationOption("ignore", "ignore", 0);
-            ui.options.push(ignoreOption);
+        /**
+         * Select the ignore option, and toggle the ignoreAlways setting if it's
+         * already selected
+         */
+        ctrl.onEventIgnore = function() {
+            if (!ctrl.isSelected) {
+                return;
+            }
+            var id = line.id + "-ignore";
+            if (id !== $scope.selectedOption.id) {
+                $scope.selectedOption.id = id;
+            } else {
+                $scope.selectedOption.ignoreAlways = !$scope.selectedOption.ignoreAlways;
+            }
+            $scope.optionChanged();
+        };
 
-            if (ui.best === "ignore") {
-                ui.best = ignoreOption;
-                ignoreOption.always = true;
-            } else if (ui.best && !ui.best.hasBug) {
-                // If we have a best option with no bug, the UI won't display a
-                // radio option for that case, set the default to the following option
-                ui.selectedOptionIndex = 1;
+        /**
+         * Select a specified options
+         * @param {string} option - numeric id of the option to select or '=' to select the
+                                    manual option
+         */
+        ctrl.onEventSelectOption = function(option) {
+            if (!ctrl.isSelected || !ctrl.isEditable) {
+                return;
+            }
+            var id;
+            if (option === "manual") {
+                id = line.id + "-manual";
+            } else {
+                var idx = parseInt(option);
+                var selectableOptions = $scope.options.filter((option) => option.selectable);
+                if (selectableOptions[idx]) {
+                    id = selectableOptions[idx].id;
+                }
+            }
+            if (!optionsById.has(id)) {
+                return;
             }
-            if (ui.best) {
-                ui.best.isBest = true;
+            if (id !== $scope.selectedOption.id) {
+                $scope.selectedOption.id = id;
+                $scope.optionChanged();
             }
+            if (option === "=") {
+                $("#" + line.id + "-manual-bug").focus();
```

Comment:
Kind of a bummer to be using raw jquery here, but I assume you couldn't find an alternative.